xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-default/src/main/java/org/xwiki/notifications/filters/migration/WatchListObjectsRemovalTaskConsumer.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.filters.migration;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.apache.commons.lang3.StringUtils;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.index.IndexException;
+import org.xwiki.index.TaskConsumer;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.LocalDocumentReference;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Task for removing WatchListClass xobjects from page. This task should only be triggered when
+ * {@link R160000000XWIKI17243DataMigration} has been done.
+ * Note that this task might also create an autowatch xobject to get the new property if it's not defined, as we use
+ * to fallback on the autowatch property defined in the WatchListClass xobject.
+ *
+ * @version $Id$
+ * @since 16.0.0RC1
+ */
+@Component
+@Named(WatchListObjectsRemovalTaskConsumer.TASK_NAME)
+@Singleton
+public class WatchListObjectsRemovalTaskConsumer implements TaskConsumer
+{
+    static final String TASK_NAME = "watchlist-xobject-removal";
```

Comment:
Maybe prefix this with the name of the module `notifications-watchlist-xobject-removal`?

Reply:
I sure can, but to my knowledge we don't really have a best practice for those hints / names?

Reply:
yes!